src/cr/cube/matrix/measure.py

```diff
@@ -1326,6 +1336,178 @@ def blocks(self):
         return ZscoreSubtotals.blocks(self._weighted_cube_counts, self._dimensions)
 
 
+# === MARGINALS ===
+
+
+class _BaseSecondOrderMarginal(object):
+    """Base class for all second-order measure objects."""
+
+    def __init__(self, dimensions, second_order_measures, cube_measures, orientation):
+        self._dimensions = dimensions
+        self._second_order_measures = second_order_measures
+        self._cube_measures = cube_measures
+        if orientation not in ("rows", "columns"):
+            raise ValueError(
+                "'%s' is an invalid orientation, must be 'rows' or 'columns'."
+                % orientation
+            )
+        self._orientation = orientation
+
+    @lazyproperty
+    def blocks(self):
+        """List of the 2 1D ndarray "blocks" making up this measure.
+
+        These are the base-values and the subtotals.
+        """
+        raise NotImplementedError(  # pragma: no cover
+            "%s must implement `.blocks`" % type(self).__name__
+        )
+
+    @lazyproperty
+    def orientation(self):
+        """String, either "row" or "column", the orientation the marginal."""
+        return self._orientation
```

Comment:
Let's make this an Enum. I'm thinking `cr.cube.enums.ORIENTATION`. I bet we end up using this more than we initially expected.